helpers/limits.go

```diff
@@ -0,0 +1,18 @@
+package helpers
+
+import (
+	"github.com/pbnjay/memory"
+)
+
+const GB_IN_BYTES uint64 = 1024 * 1024 * 1024
+const GO_ROUTINES_PER_GB uint64 = 250000
+
+func GetMaxGoRoutines() uint64 {
```

Comment:
Remove TODOs about int64 conversions, make this one return int64
If it's > uint32 cap it to maxint64

Reply:
Fixed it. `GetMaxGoRoutines` should return uint64 as this is what the memory can return. (it can't have negative values). The check that was missing is near the passing to the semaphore (as semaphore can have negative value) and thus we need to check that the value is not more then `MAX_INT64` and if yes just cap it at MAX_INT64 but this function should be left as is I believe.

Reply:
Yes but blindly casting uint64's tied to memory and multipliers to int64 won't work: https://go.dev/play/p/KtA-WjWeFcQ